crates/bevy_ecs/src/change_detection.rs

```diff
@@ -103,7 +103,10 @@ pub trait DetectChangesMut: DetectChanges {
     /// **Note**: This operation cannot be undone.
     fn set_changed(&mut self);
 
-    /// Manually sets the change tick recording the previous time this data was mutated.
+    /// Manually sets the change tick recording the previous time this data's changed status was checked.
```

Comment:
This isn't what we want this method to do: the bug in this method should be fixed as part of this PR.

We may *also* want a method for the previous functionality (last_checked perhaps), but manually checking and messing with "when was this data last changed" was the intent of the feature request.

Reply:
I'm gonna second Alice here. I've always wondered why these methods used `last_change_tick` instead of ` -- I never realized it was just a mistake.

Reply:
Updated `set_last_changed`, and the changelog/migration guide.